modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,330 @@
+
+
+#define SHIBARI_TIGHTNESS_LOW (1<<0)
+#define SHIBARI_TIGHTNESS_MED (1<<1)
+#define SHIBARI_TIGHTNESS_HIGH (1<<2)
+
+/obj/item/clothing/under/shibari
+	strip_delay = 100
+	can_adjust = FALSE
+	body_parts_covered = NONE
+	mutant_variants = STYLE_DIGITIGRADE|STYLE_TAUR_ALL
+	item_flags = DROPDEL
+	greyscale_colors = "#bd8fcf"
+
+	///Tightness of the ropes can be low, medium and hard. This var works as multiplier for arousal and pleasure recieved while wearing this item
+	var/tightness = SHIBARI_TIGHTNESS_LOW
+	///Rope amount yielded from this apparel
+	var/rope_amount = 1
+
+	///should this clothing item use the emissive system
+	var/glow = FALSE
+
+/obj/item/clothing/under/shibari/update_overlays()
+	. = ..()
+	if(glow)
+		. += emissive_appearance(icon, icon_state, alpha = alpha)
+
+/obj/item/clothing/under/shibari/worn_overlays(mutable_appearance/standing, isinhands, icon_file)
+	. = ..()
+	if(glow)
+		. += emissive_appearance(standing.icon, standing.icon_state, alpha = standing.alpha)
+
+/obj/item/clothing/under/shibari/Destroy(force)
+	for(var/obj/item in contents)
+		item.forceMove(get_turf(src))
+	var/mob/living/carbon/human/hooman = loc
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.remove_status_effect(/datum/status_effect/ropebunny)
+	. = ..()
+
+/obj/item/clothing/under/shibari/equipped(mob/user, slot)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_ATTACK_HAND, .proc/handle_take_off, user)
+
+
+/obj/item/clothing/under/shibari/proc/handle_take_off(datum/source, mob/user)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/handle_take_off_async, user)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/obj/item/clothing/under/shibari/proc/handle_take_off_async(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/human/hooman = user
+		if(do_after(hooman, HAS_TRAIT(hooman, TRAIT_RIGGER) ? 2 SECONDS : 10 SECONDS, target = src))
+			qdel(src)
+
+/obj/item/clothing/under/shibari/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/clothing/under/shibari/AltClick(mob/user)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(user != hooman)
+		switch(tightness)
+			if(SHIBARI_TIGHTNESS_LOW)
+				tightness = SHIBARI_TIGHTNESS_MED
+			if(SHIBARI_TIGHTNESS_MED)
+				tightness = SHIBARI_TIGHTNESS_HIGH
+			if(SHIBARI_TIGHTNESS_HIGH)
+				tightness = SHIBARI_TIGHTNESS_LOW
+
+/obj/item/clothing/under/shibari/process(delta_time)
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_LOW && hooman.arousal < 15)
```

Comment:
There's no preference checking going on here.

Reply:
Also, descriptive variable names, please.

Reply:
You cannot apply shibari on character without prefs anyway, additional lines are not required. Descriptive vars - sure thing